sensus/sensus/code.py

```diff
@@ -1,28 +1,91 @@
-        
-    def sigmoid(self, z): 
-        """ The output of a neuron """
-        return 1 / (1 + np.exp(-z)) # the sigmoid function transforms the output into a number between 0 and 1, which becomes
-                                 # input to the next layer
+        self.biases = [np.random.randn(j, 1) for j in sizes[1:]] # a vector of initial biases, excluding the input layer
     
-    def activation(self, a): # a is the vector of outputs from the previous layer (i.e., input of the current layer)
+    # The activation function that computes the input of a neuron as the weighted sum of the previous neurons' outputs, 
+    # the weights and the biases
+    
+    def activation(self, a): # 'a' is a vector of outputs from the previous layer (i.e., input of the current layer)
         """ The activation of a neuron """
         for w, b in zip(self.weights, self.biases):
-            z = w * a + b # the activations of the neurons in the previous layer (i.e., inputs) are multiplied by the 
-                          # weights and added to the biases of the neurons in the current layer
+            z = np.dot(w, a) + b 
         return z
     
+    # The 'activation function' sigmoid, which compresses the activation value to a value between 0 and 1.
+    
+    def sigmoid(self, z): 
+        """ The output of a neuron """
+        return 1 / (1 + np.exp(-z))
+   
+    # We enter the activation value of the neurons into the sigmoid function to get the outputs. 
+    # This specific function will compute the output of the neural network. 
+    
     def output(self, a):
         return sigmoid(activation(a))
     
+    ## The following functions will enable the output function to give a correct value. Our goal is to find the weights
+    ## and the biases that will lead to the correct output. To find this, we need to find how much the cost function 
+    ## (i.e., the difference between the network's output and the correct output, as provided by the training data) 
+    ## would change based on changes in the weights and the biases. 
+    
+    # We start by finding the change in the cost function based on the changes in the output activations. 
+    # How less wrong would we be if the final output activations were different? This information is the first 
+    # step towards finding the change in the cost function for weights and biases, because it will be communicated
+    # back to the previous layers by 'backpropagation'. 
+    
+    def cost_output_derivative(self, output_activations, y): # output_activations are created in backpropagation
+        """ Returns the partial derivates of the cost function with respect to all the output activations. """
+        return output_activations - y
+    
+    def sigmoid_prime(z): # multiplying this with the derivative of the cost function we get an error term, which 
+                          # tells us how much the output would change depending on the output activations. 
```

Comment:
def sigmoid_prime(self, z):

And then also change return self.signmoid(z) * 1- self.sigmoid(z))